connectors/sources/mongo.py

```diff
@@ -196,14 +198,16 @@ def get_client(self):
             else:
                 client_params["tls"] = False
 
-            client = AsyncIOMotorClient(self.host, **client_params)
-
-            db = client[self.configuration["database"]]
-            self.collection = db[self.configuration["collection"]]
```

Comment:
Fun problem from design perspective:

This is a context manager, but it not only yields the `client`, but also sets up instance variable `self.collection` - looks like a logical leak that should not be there